<?xml version='1.0' encoding='UTF-8'?>
<root>
 <tabbed_widget parent="main_window" name="Main Window">
  <Tab tab_name="base_pose" containers="1">
   <Container>
    <DockSplitter sizes="1" count="1" orientation="-">
     <DockSplitter sizes="0.333333;0.333333;0.333333" count="3" orientation="|">
      <DockSplitter sizes="0.5;0.5" count="2" orientation="-">
       <DockArea name="...">
        <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
         <range top="1.981667" right="313.920000" bottom="-0.048333" left="283.920000"/>
         <limitY/>
         <curve name="/multirotor_controller/base_state/pose/pos/x" color="#1f77b4"/>
         <curve name="/multirotor_controller/command/target_position/x" color="#1f77b4"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
         <range top="0.001768" right="313.920000" bottom="-0.000704" left="283.920000"/>
         <limitY/>
         <curve name="/multirotor_controller/base_state/pose/rpy/x" color="#1f77b4"/>
         <curve name="/multirotor_controller/feedback/desired_roll" color="#ff7f0e"/>
        </plot>
       </DockArea>
      </DockSplitter>
      <DockSplitter sizes="0.5;0.5" count="2" orientation="-">
       <DockArea name="...">
        <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
         <range top="0.008720" right="313.920000" bottom="-0.000213" left="283.920000"/>
         <limitY/>
         <curve name="/multirotor_controller/base_state/pose/pos/y" color="#d62728"/>
         <curve name="/multirotor_controller/command/target_position/y" color="#d62728"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
         <range top="0.103064" right="313.920000" bottom="-0.039780" left="283.920000"/>
         <limitY/>
         <curve name="/multirotor_controller/base_state/pose/rpy/y" color="#d62728"/>
         <curve name="/multirotor_controller/feedback/desired_pitch" color="#f14cc1"/>
        </plot>
       </DockArea>
      </DockSplitter>
      <DockSplitter sizes="0.5;0.5" count="2" orientation="-">
       <DockArea name="...">
        <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
         <range top="4.211667" right="313.920000" bottom="0.921667" left="283.920000"/>
         <limitY/>
         <curve name="/multirotor_controller/base_state/pose/pos/z" color="#1ac938"/>
         <curve name="/multirotor_controller/command/target_position/z" color="#1ac938"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
         <range top="0.000045" right="313.920000" bottom="-0.000003" left="283.920000"/>
         <limitY/>
         <curve name="/multirotor_controller/base_state/pose/rpy/z" color="#ff7f0e"/>
         <curve name="/multirotor_controller/command/target_yaw_angle" color="#f14cc1"/>
        </plot>
       </DockArea>
      </DockSplitter>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="base_twist" containers="1">
   <Container>
    <DockSplitter sizes="1" count="1" orientation="-">
     <DockSplitter sizes="0.333333;0.333333;0.333333" count="3" orientation="|">
      <DockSplitter sizes="0.5;0.5" count="2" orientation="-">
       <DockArea name="...">
        <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
         <range top="0.800864" right="313.900000" bottom="-0.019534" left="283.930000"/>
         <limitY/>
         <curve name="/multirotor_controller/base_state/twist/linear/x" color="#f14cc1"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
         <range top="0.001532" right="313.900000" bottom="-0.000853" left="283.930000"/>
         <limitY/>
         <curve name="/multirotor_controller/base_state/twist/angular/x" color="#bcbd22"/>
        </plot>
       </DockArea>
      </DockSplitter>
      <DockSplitter sizes="0.5;0.5" count="2" orientation="-">
       <DockArea name="...">
        <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
         <range top="0.006044" right="313.900000" bottom="-0.002445" left="283.930000"/>
         <limitY/>
         <curve name="/multirotor_controller/base_state/twist/linear/y" color="#9467bd"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
         <range top="0.154422" right="313.900000" bottom="-0.180011" left="283.930000"/>
         <limitY/>
         <curve name="/multirotor_controller/base_state/twist/angular/y" color="#1f77b4"/>
        </plot>
       </DockArea>
      </DockSplitter>
      <DockSplitter sizes="0.5;0.5" count="2" orientation="-">
       <DockArea name="...">
        <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
         <range top="1.129602" right="313.900000" bottom="-0.986922" left="283.930000"/>
         <limitY/>
         <curve name="/multirotor_controller/base_state/twist/linear/z" color="#17becf"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
         <range top="0.000037" right="313.900000" bottom="-0.000087" left="283.930000"/>
         <limitY/>
         <curve name="/multirotor_controller/base_state/twist/angular/z" color="#d62728"/>
        </plot>
       </DockArea>
      </DockSplitter>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="motor_speed" containers="1">
   <Container>
    <DockSplitter sizes="1" count="1" orientation="-">
     <DockSplitter sizes="0.5;0.5" count="2" orientation="|">
      <DockSplitter sizes="0.5;0.5" count="2" orientation="-">
       <DockArea name="...">
        <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
         <range top="690.730520" right="313.900000" bottom="626.664721" left="283.940000"/>
         <limitY/>
         <curve name="/iris/command/motor_speed/angular_velocities.0" color="#1f77b4"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
         <range top="690.730515" right="313.900000" bottom="626.664740" left="283.940000"/>
         <limitY/>
         <curve name="/iris/command/motor_speed/angular_velocities.2" color="#1ac938"/>
        </plot>
       </DockArea>
      </DockSplitter>
      <DockSplitter sizes="0.5;0.5" count="2" orientation="-">
       <DockArea name="...">
        <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
         <range top="690.730495" right="313.900000" bottom="626.664718" left="283.940000"/>
         <limitY/>
         <curve name="/iris/command/motor_speed/angular_velocities.1" color="#d62728"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
         <range top="690.730501" right="313.900000" bottom="626.664714" left="283.940000"/>
         <limitY/>
         <curve name="/iris/command/motor_speed/angular_velocities.3" color="#9467bd"/>
        </plot>
       </DockArea>
      </DockSplitter>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <Tab tab_name="feedback" containers="1">
   <Container>
    <DockSplitter sizes="0.333071;0.333857;0.333071" count="3" orientation="-">
     <DockSplitter sizes="0.333333;0.333333;0.333333" count="3" orientation="|">
      <DockArea name="...">
       <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
        <range top="1.011325" right="313.900000" bottom="-0.389880" left="283.940000"/>
        <limitY/>
        <curve name="/multirotor_controller/feedback/desired_acceleration/x" color="#1f77b4"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
        <range top="0.000419" right="313.900000" bottom="-0.017179" left="283.940000"/>
        <limitY/>
        <curve name="/multirotor_controller/feedback/desired_acceleration/y" color="#d62728"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
        <range top="0.819364" right="313.900000" bottom="-1.060742" left="283.940000"/>
        <limitY/>
        <curve name="/multirotor_controller/feedback/desired_acceleration/z" color="#1ac938"/>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter sizes="0.251156;0.248844;0.251156;0.248844" count="4" orientation="|">
      <DockArea name="...">
       <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
        <range top="16.311143" right="313.900000" bottom="13.425144" left="283.940000"/>
        <limitY/>
        <curve name="/multirotor_controller/feedback/thrust_force_sum" color="#9467bd"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
        <range top="0.001752" right="313.900000" bottom="-0.000043" left="283.940000"/>
        <limitY/>
        <curve name="/multirotor_controller/feedback/desired_roll" color="#ff7f0e"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
        <range top="0.103064" right="313.900000" bottom="-0.039780" left="283.940000"/>
        <limitY/>
        <curve name="/multirotor_controller/feedback/desired_pitch" color="#f14cc1"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
        <range top="0.100000" right="313.900000" bottom="-0.100000" left="283.940000"/>
        <limitY/>
        <curve name="/multirotor_controller/feedback/desired_yaw" color="#17becf"/>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter sizes="0.251156;0.248844;0.251156;0.248844" count="4" orientation="|">
      <DockArea name="...">
       <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
        <range top="4.077786" right="313.900000" bottom="3.356286" left="283.940000"/>
        <limitY/>
        <curve name="/multirotor_controller/feedback/thrust_forces.0" color="#bcbd22"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
        <range top="4.077786" right="313.900000" bottom="3.356286" left="283.940000"/>
        <limitY/>
        <curve name="/multirotor_controller/feedback/thrust_forces.1" color="#1f77b4"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
        <range top="4.077786" right="313.900000" bottom="3.356286" left="283.940000"/>
        <limitY/>
        <curve name="/multirotor_controller/feedback/thrust_forces.2" color="#d62728"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot mode="TimeSeries" flip_y="false" flip_x="false" style="Lines">
        <range top="4.077786" right="313.900000" bottom="3.356286" left="283.940000"/>
        <limitY/>
        <curve name="/multirotor_controller/feedback/thrust_forces.3" color="#1ac938"/>
       </plot>
      </DockArea>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <currentTabIndex index="0"/>
 </tabbed_widget>
 <use_relative_time_offset enabled="0"/>
 <!-- - - - - - - - - - - - - - - -->
 <!-- - - - - - - - - - - - - - - -->
 <Plugins>
  <plugin ID="DataLoad CSV">
   <default time_axis="" delimiter="0"/>
  </plugin>
  <plugin ID="DataLoad MCAP"/>
  <plugin ID="DataLoad ROS bags">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
  </plugin>
  <plugin ID="DataLoad ULog"/>
  <plugin ID="ROS Topic Subscriber">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
   <boolean_strings_to_number value="true"/>
   <remove_suffix_from_strings value="true"/>
  </plugin>
  <plugin ID="UDP Server"/>
  <plugin ID="WebSocket Server"/>
  <plugin ID="ZMQ Subscriber"/>
  <plugin ID="Fast Fourier Transform"/>
  <plugin ID="Quaternion to RPY"/>
  <plugin ID="Reactive Script Editor">
   <library code="--[[ Helper function to create a series from arrays&#xa;&#xa; new_series: a series previously created with ScatterXY.new(name)&#xa; prefix:     prefix of the timeseries, before the index of the array&#xa; suffix_X:   suffix to complete the name of the series containing the X value. If [nil], use the index of the array.&#xa; suffix_Y:   suffix to complete the name of the series containing the Y value&#xa; timestamp:   usually the tracker_time variable&#xa;              &#xa; Example:&#xa; &#xa; Assuming we have multiple series in the form:&#xa; &#xa;   /trajectory/node.{X}/position/x&#xa;   /trajectory/node.{X}/position/y&#xa;   &#xa; where {N} is the index of the array (integer). We can create a reactive series from the array with:&#xa; &#xa;   new_series = ScatterXY.new(&quot;my_trajectory&quot;) &#xa;   CreateSeriesFromArray( new_series, &quot;/trajectory/node&quot;, &quot;position/x&quot;, &quot;position/y&quot;, tracker_time );&#xa;]]--&#xa;&#xa;function CreateSeriesFromArray( new_series, prefix, suffix_X, suffix_Y, timestamp )&#xa;  &#xa;  --- clear previous values&#xa;  new_series:clear()&#xa;  &#xa;  --- Append points to new_series&#xa;  index = 0&#xa;  while(true) do&#xa;&#xa;    x = index;&#xa;    -- if not nil, get the X coordinate from a series&#xa;    if suffix_X ~= nil then &#xa;      series_x = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_X) )&#xa;      if series_x == nil then break end&#xa;      x = series_x:atTime(timestamp)&#x9; &#xa;    end&#xa;    &#xa;    series_y = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_Y) )&#xa;    if series_y == nil then break end &#xa;    y = series_y:atTime(timestamp)&#xa;    &#xa;    new_series:push_back(x,y)&#xa;    index = index+1&#xa;  end&#xa;end&#xa;&#xa;--[[ Similar to the built-in function GetSeriesNames(), but select only the names with a give prefix. ]]--&#xa;&#xa;function GetSeriesNamesByPrefix(prefix)&#xa;  -- GetSeriesNames(9 is a built-in function&#xa;  all_names = GetSeriesNames()&#xa;  filtered_names = {}&#xa;  for i, name in ipairs(all_names)  do&#xa;    -- check the prefix&#xa;    if name:find(prefix, 1, #prefix) then&#xa;      table.insert(filtered_names, name);&#xa;    end&#xa;  end&#xa;  return filtered_names&#xa;end&#xa;&#xa;--[[ Modify an existing series, applying offsets to all their X and Y values&#xa;&#xa; series: an existing timeseries, obtained with TimeseriesView.find(name)&#xa; delta_x: offset to apply to each x value&#xa; delta_y: offset to apply to each y value  &#xa;&#xa;]]--&#xa;&#xa;function ApplyOffsetInPlace(series, delta_x, delta_y)&#xa;  -- use C++ indeces, not Lua indeces&#xa;  for index=0, series:size()-1 do&#xa;    x,y = series:at(index)&#xa;    series:set(index, x + delta_x, y + delta_y)&#xa;  end&#xa;end&#xa;"/>
   <scripts/>
  </plugin>
  <plugin ID="CSV Exporter"/>
  <plugin ID="ROS /rosout Visualization"/>
  <plugin ID="ROS Topic Re-Publisher"/>
 </Plugins>
 <!-- - - - - - - - - - - - - - - -->
 <previouslyLoaded_Datafiles/>
 <previouslyLoaded_Streamer name="ROS Topic Subscriber"/>
 <!-- - - - - - - - - - - - - - - -->
 <customMathEquations/>
 <snippets/>
 <!-- - - - - - - - - - - - - - - -->
</root>

